app/providers/atmosphere/local_pdp.rb

```diff
@@ -0,0 +1,54 @@
+#
+# Local PDP with basic service sharing functionality.
+# Does not require external services to operate.
+#
+module Atmosphere
+  class LocalPdp
+    def initialize(current_user)
+      @current_user = current_user
+    end
+
+    def can_start_in_production?(at)
+      Rails.logger.debug("can_start_in_production: using local PDP")
+      uat = UserApplianceType.where(
+        user: @current_user,
+        appliance_type: at
+      )
+      uat.present?
+    end
+
+    def can_start_in_development?(at)
+      Rails.logger.debug("can_start_in_development: using local PDP")
+      uat = UserApplianceType.where(
+          user: @current_user,
+          appliance_type: at
+      )
+      uat.present? and ['developer', 'manager'].include? uat.first.role
+    end
+
+    def can_manage?(obj)
+      uat = UserApplianceType.where(
+          user: @current_user,
+          appliance_type: obj,
+          role: 'manager'
+      )
+      obj.user_id == current_user.id or uat.present?
+    end
+
+    def filter(ats, filter = nil)
+      ats.where(visibility_for_filter(filter.to_s))
+    end
+
+    private
+
+    attr_reader :current_user
+
+    def visibility_for_filter(filter)
```

Comment:
It seems like copy paste from `DefaultPdp`. Instead `UserApplianceType` should be used to check which ATs are visible.